app/react/Templates/components/MetadataTemplate.tsx

```diff
@@ -35,12 +35,12 @@ interface MetadataTemplateProps {
   _id?: string;
 }
 
-const getTemplateDefaultColor = (allTemplates: List<TemplateSchema>, templateId: string) => {
-  if (!templateId) {
-    return COLORS[allTemplates.size % COLORS.length];
+const getTemplateDefaultColor = (allTemplates: List<TemplateSchema>, template: any) => {
+  let color = COLORS[allTemplates.size % COLORS.length];
+  if (template.data.color) {
```

Comment:
Just as a very small thing, maybe you could invert the order of the condition to avoid calculating the color when it is not needed.

Reply:
Refactored as suggested.